integration-testing/build.gradle

```diff
@@ -68,12 +69,12 @@ task mavenTest(type: Test) {
     def mavenTestClasspathConfiguration = project.configurations.detachedConfiguration(
             project.dependencies.create("org.jetbrains.kotlinx:kotlinx-coroutines-core:$version"),
             project.dependencies.create("org.jetbrains.kotlinx:kotlinx-coroutines-android:$version"))
-
     mavenTestClasspathConfiguration.attributes {
         attribute(KotlinPlatformType.attribute, KotlinPlatformType.jvm)
     }
-
-    classpath += mavenTestClasspathConfiguration
+    /* the order is this way because the first match is used, and if we ever accidentally add the wrong dependencies to
+    this source-set, they shouldn't override the dependencies to the local Maven repository */
+    classpath = mavenTestClasspathConfiguration + classpath
```

Comment:
With this, I was unable to make the new check fail, even trying to specify the dependencies for this source set in various places.